File: NetFramework/WinForms/CS/TouchDemo/TouchEvent/AutoRepeatDemo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace TouchToolKitDemo.TouchEvent
{
    public partial class AutoRepeatDemo : DemoBase
    {
        int number1 = 99;
        int number2 = 99;
        Timer timer;
        int timerInterval;
        Control _targetButton = null;

        public AutoRepeatDemo()
        {
            InitializeComponent();
            InitializeDemo();
        }

        private void InitializeDemo()
        {
            this.Title = "AutoRepeat";
            this.Description = "Provide your own spin buttons by using Buttons with C1TouchEventProvider's EnablePressAndHold.\n    -Try to press and hold on the spin button to increase and decrease the number.";
            
            textBox1.Text = number1.ToString();
            textBox2.Text = number2.ToString();
        }

        private void InputTextBox_Validated(object sender, EventArgs e)
        {
            TextBox textbox = sender as TextBox;
            if (textbox != null)
            {
                try
                {
                    if (textbox == textBox1)
                    {
                        number1 = Convert.ToInt32(textbox.Text);
                    }
                    else if (textbox == textBox2)
                    {
                        number2 = Convert.ToInt32(textbox.Text);
                    }
                }
                catch (FormatException)
                {
                    if (textbox == textBox1)
                    {
                        textbox.Text = number1.ToString();
                    }
                    else if (textbox == textBox2)
                    {
                        textbox.Text = number2.ToString();
                    }
                }
            }
        }

        private void upButton2_Click(object sender, EventArgs e)
        {
            number2++;
            textBox2.Text = number2.ToString();
        }

        private void downButton2_Click(object sender, EventArgs e)
        {
            number2--;
            textBox2.Text = number2.ToString();
        }

        private void C1TouchEventProvider1_Tapped(object sender, C1.Win.TouchToolKit.TappedEventArgs e)
        {
            _targetButton = e.TargetControl;
            UpdateValueBySpinButon();
        }

        private void UpdateValueBySpinButon()
        {
            if (_targetButton == upButton1)
            {
                number1++;
                textBox1.Text = number1.ToString();
            }
            else if (_targetButton == downButton1)
            {
                number1--;
                textBox1.Text = number1.ToString();
            }
        }

        private void C1TouchEventProvider1_PointerPressed(object sender, C1.Win.TouchToolKit.PointerEventArgs e)
        {
            _targetButton = e.TargetControl;
            StartTimer();
        }

        private void C1TouchEventProvider1_PointerReleased(object sender, C1.Win.TouchToolKit.PointerEventArgs e)
        {
            _targetButton = null;
            StopTimer();
        }

        private void StartTimer()
        {
            if (timer == null)
            {
                timer = new Timer();
                timer.Tick += timer_Tick;
            }
            timerInterval = 500;
            timer.Interval = timerInterval;
            timer.Start();
        }

        private void StopTimer()
        {
            if (timer != null)
            {
                timer.Stop();
                timer.Dispose();
                timer = null;
            }
        }

        private void timer_Tick(object sender, EventArgs e)
        {
            UpdateValueBySpinButon();

            timerInterval *= 7;
            timerInterval /= 10;
            if (timerInterval < 1)
            {
                timerInterval = 1;
            }
            timer.Interval = timerInterval;
        }
    }
}

File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/TouchEvent/AutoRepeatDemo.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace TouchEvent
    Partial Public Class AutoRepeatDemo
        Inherits DemoBase
        Private number1 As Integer = 99
        Private number2 As Integer = 99
        Private timer As Timer
        Private timerInterval As Integer
        Private _targetButton As Control = Nothing

        Public Sub New()
            InitializeComponent()
            InitializeDemo()
        End Sub

        Private Sub InitializeDemo()
            Me.Title = "AutoRepeat"
            Me.Description = "Provide your own spin buttons by using Buttons with EnablePressAndHold of C1TouchEventProvider." & vbLf & "    -Try to press and hold on the spin button to increase and decrease the number."

            textBox1.Text = number1.ToString()
            textBox2.Text = number2.ToString()
        End Sub

        Private Sub InputTextBox_Validated(sender As Object, e As EventArgs)
            Dim textboxTmp As TextBox = TryCast(sender, TextBox)
            If textboxTmp IsNot Nothing Then
                Try
                    If Object.ReferenceEquals(textboxTmp, textBox1) Then
                        number1 = Convert.ToInt32(textboxTmp.Text)
                    ElseIf Object.ReferenceEquals(textboxTmp, textBox2) Then
                        number2 = Convert.ToInt32(textboxTmp.Text)
                    End If
                Catch generatedExceptionName As FormatException
                    If Object.ReferenceEquals(textboxTmp, textBox1) Then
                        textboxTmp.Text = number1.ToString()
                    ElseIf Object.ReferenceEquals(textboxTmp, textBox2) Then
                        textboxTmp.Text = number2.ToString()
                    End If
                End Try
            End If
        End Sub

        Private Sub upButton2_Click(sender As Object, e As EventArgs)
            number2 += 1
            textBox2.Text = number2.ToString()
        End Sub

        Private Sub downButton2_Click(sender As Object, e As EventArgs)
            number2 -= 1
            textBox2.Text = number2.ToString()
        End Sub

        Private Sub gcTouchEventProvider1_Tapped(sender As Object, e As C1.Win.TouchToolKit.TappedEventArgs) Handles gcTouchEventProvider1.Tapped
            _targetButton = e.TargetControl
            UpdateValueBySpinButon()
        End Sub

        Private Sub UpdateValueBySpinButon()
            If Object.ReferenceEquals(_targetButton, upButton1) Then
                number1 += 1
                textBox1.Text = number1.ToString()
            ElseIf Object.ReferenceEquals(_targetButton, downButton1) Then
                number1 -= 1
                textBox1.Text = number1.ToString()
            End If
        End Sub

        Private Sub gcTouchEventProvider1_PointerPressed(sender As Object, e As C1.Win.TouchToolKit.PointerEventArgs) Handles gcTouchEventProvider1.PointerPressed
            _targetButton = e.TargetControl
            StartTimer()
        End Sub

        Private Sub gcTouchEventProvider1_PointerReleased(sender As Object, e As C1.Win.TouchToolKit.PointerEventArgs) Handles gcTouchEventProvider1.PointerReleased
            _targetButton = Nothing
            StopTimer()
        End Sub

        Private Sub StartTimer()
            If timer Is Nothing Then
                timer = New Timer()
                AddHandler timer.Tick, AddressOf timer_Tick
            End If
            timerInterval = 500
            timer.Interval = timerInterval
            timer.Start()
        End Sub

        Private Sub StopTimer()
            If timer IsNot Nothing Then
                timer.[Stop]()
                timer.Dispose()
                timer = Nothing
            End If
        End Sub

        Private Sub timer_Tick(sender As Object, e As EventArgs)
            UpdateValueBySpinButon()

            timerInterval *= 7
            timerInterval /= 10
            If timerInterval < 1 Then
                timerInterval = 1
            End If
            timer.Interval = timerInterval
        End Sub
    End Class
End Namespace
